hybridse/src/udf/udf.cc

```diff
@@ -88,6 +94,24 @@ int32_t year(int64_t ts) {
     return t.tm_year + 1900;
 }
 
+int32_t dayofyear(codec::Timestamp *ts) { return dayofyear(ts->ts_); }
+int32_t dayofyear(codec::Date *date) {
+    int32_t day, month, year;
+    if (!codec::Date::Decode(date->date_, &year, &month, &day)) {
+        return 0;
+    }
+    try {
+        if (month <= 0 || month > 12) {
+            return 0;
+        } else if (day <= 0 || day > 31) {
+            return 0;
+        }
+        boost::gregorian::date d(year, month, day);
```

Comment:
Try remove dependency for boost. We'd like remove boost from our third party list in the further, boost compiling is pain

Reply:
`boost::gregorian` has been used by `dayofyear()`, `weekofyear()`, `date_format()`, `string_to_date()`, etc.
So maybe we can create another issue to remove boost dependencies and keep using `boost::gregorian` in this pr.

Reply:
Both fine with that ~

Reply:
I will look into removing boost dependencies in another PR as @jingchen2222 said